packages/transport/src/chrome-runtime/client-connection-manager.ts

```diff
@@ -129,20 +128,25 @@ export class ClientConnectionManager {
      */
     const serviceListener: Parameters<chrome.runtime.PortMessageEvent['addListener']>[0] = (
       ev: unknown,
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      _: chrome.runtime.Port,
     ) => {
-      if (!isTransportData(ev)) throw Error('Unknown data in transport');
-      const response = ev;
-      if (isTransportMessage(response)) {
-        const { requestId, message } = response;
-        clientPort.postMessage({ requestId, message } satisfies TransportMessage);
-      } else if (isTransportInitChannel(response)) {
-        const { requestId, channel: subName } = response;
-        const sourcePort = chrome.runtime.connect({ includeTlsChannelId: true, name: subName });
-        const stream = new ReadableStream(new ChromeRuntimeStreamSource(sourcePort));
-        clientPort.postMessage({ requestId, stream } satisfies TransportStream, [stream]);
-      } else throw Error('Unimplemented response kind');
+      try {
+        if (isTransportState(ev)) {
+          clientPort.postMessage(ev);
+          if (!ev.connected) this.endConnection(clientId);
+        } else if (isTransportData(ev)) {
+          if (isTransportMessage(ev)) clientPort.postMessage(ev);
+          else if (isTransportInitChannel(ev)) {
+            const { requestId, channel: subName } = ev;
+            const sourcePort = chrome.runtime.connect({ includeTlsChannelId: true, name: subName });
+            const stream = new ReadableStream(new ChromeRuntimeStreamSource(sourcePort));
+            clientPort.postMessage({ requestId, stream }, [stream]);
+          } else throw Error('Unimplemented response kind');
+        } else if (isTransportError(ev)) clientPort.postMessage(ev);
+        else throw Error('Unknown transport from service');
+      } catch (error) {
+        console.error('Error in service listener', error, ev);
+        clientPort.postMessage({ error: String(error) });
+      }
```

Comment:
This multi-conditional block is a bit hard to read. Not sure how to improve though 🤔